packages/skill-lesson/hooks/use-progress.ts

```diff
@@ -3,59 +3,75 @@ import {SanityDocument} from '@sanity/client'
 import {trpcSkillLessons} from '../utils/trpc-skill-lessons'
 import {first, filter, find, flatMap, get, indexOf, isEmpty, map} from 'lodash'
 import {getNextSection} from '../utils/get-next-section'
+import {LessonResource} from '../schemas/lesson-resource'
+import {LessonProgress} from '@skillrecordings/database'
+
+export type Exercise = LessonResource & {solution?: any}
+export type Module = {
+  sections: Section[]
+  _id: string
+  slug: {current: string}
+  moduleType: string
+}
+export type Section = {lessons: Exercise[]; _id: string; slug: string}
 
-export const useModuleProgress = ({module}: {module: SanityDocument}) => {
-  const {sections} = module
-  const {data: userProgress, status: userProgressStatus} =
-    trpcSkillLessons.progress.get.useQuery()
-
-  const exercises = flatMap(sections, (s) => s.lessons)
-  const resources = flatMap(sections, (s) =>
-    map(s.lessons, (l) => (l.solution ? l.solution : l)),
+export function extractExercisesAndResource(sections: Section[]) {
```

Comment:
I "extracted methods" for testing that completely avoid async/hooks/etc but allow us to test the LOGIC and creates seams